src/OVAL/oval_probe.c

```diff
@@ -498,9 +497,13 @@ void oval_probe_meta_list(FILE *output, int flags)
 		output = stdout;
 
 	probe_dir = oval_probe_ext_getdir();
-	assume_d(probe_dir != NULL, /* void */);
```

Comment:
This code will behave the same as the macro when not in debug mode. The macro would `abort()` in order to trigger the debugger. The functionality is not preserved on purpose.

Reply:
Great! If we remove every weird construction, we will have less likely the Heisenbug.

Reply:
Wait what? I am really lost here. So do we know and acknowledge that we are changing behavior with these changes? Also, why even bother changing this in the same PR that already has tons of changes? There is nothing in these assume macros that doesn't work on Windows.

Please, please, please, if we decide to do something about this let's do it in a separate PR with a nice description of what is happening. This seems like a change that doesn't belong here.

Reply:
@mpreisler I am sorry, I don't understand at all what do you want. What do you suggest to change?

Reply:
@mpreisler I think it was not clear why I replace these macros. The problem is that Visual Studio 2017 can't compile them. Visual Studio 2017 fails and produces errors. It complains either that it expects an expression or doesn't expect an operator when expanding these macros. I found it more understandable and beneficial to replace them by normal code than wrapping them into `#ifndef _WIN32 ... #endif` sections. Is it OK?

Reply:
without looking more into it I am not sure but what exactly does it complain about? I'd suggest to fix the macros instead of removing them? also consider using commit message such as "replaced assume_d macros because they cause build issues in MSVC2017..." instead of  "replaced weird macros with normal code"

Reply:
@mpreisler 

For example, in this particular case that we comment, `oval_probe.c`:

```
Error (active)	E0029	expected an expression	oval_object	c:\devel\openscap\src\OVAL\oval_probe.c	505		

Warning	C4003	not enough actual parameters for macro 'assume'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Warning	C4003	not enough actual parameters for macro '__LB'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Warning	C4003	not enough actual parameters for macro '__terminate'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Warning	C4003	not enough actual parameters for macro '__RB'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Error	C2059	syntax error: ')'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Error	C2065	'__cont505': undeclared identifier	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Error	C2059	syntax error: 'else'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Error	C2059	syntax error: '}'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	505		

Error	C2065	'probe_dir': undeclared identifier	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	507		

```

And then about 10 errors in the rest of the file, because this weird macro causes that Microsoft Visual Studio 2017 cannot understand the rest of file. For example:

```
Error	C2099	initializer is not a constant	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	507		

Error	C2059	syntax error: 'if'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	508		

Error	C2059	syntax error: 'for'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	512		

Error	C2143	syntax error: missing '{' before '<'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	512		

Error	C2059	syntax error: '<'	oval_object	C:\devel\openscap\src\OVAL\oval_probe.c	512		
```
and so on.

When I replace the macro by `if(...)` as I propose in this patch, all of those errors and warnings disappear.

A very similar situation is at all other occurrences of assume macro.

Reply:
I briefly looked at it and assume is a variadic macro. We don't use that aspect of it almost anywhere so can't you just make it non-variadic? It should be a pretty quick fix in `src/common/assume.h`.